test/tracecontext/tracestate.js

```diff
@@ -0,0 +1,289 @@
+
+  t.ok(!t1.setValue('foo-toolong', tooBig), 'failed to set value > 256 chars')
+  t.ok(!t1.setValue(tooBig, 'foo'), 'failed to set key > 256 chars')
+
+  t.ok(t1.setValue('foo', 'bar'), 'set valid key and value')
+
+  t.ok(!t1.getValue('f:oo'), 'did not set invalid key f:oo')
+  t.ok(!t1.getValue('f;oo'), 'did not set invalid key f;oo')
+  t.ok(!t1.getValue(tooBig), 'did not set super long key')
+  t.ok(!t1.getValue('foo-colon'), 'did not set invalid value for foo-colon key')
+  t.ok(!t1.getValue('foo-semicolon'), 'did not set invalid value for foo-semicolon key')
+  t.ok(!t1.getValue('foo-comma'), 'did not set invalid value for foo-comma key')
+  t.ok(!t1.getValue('foo-equals'), 'did not set invalid value for foo-equals key')
+  t.ok(!t1.getValue('foo-toolong'), 'did not set invalid value for foo-toolong key')
+
+  t.equals(t1.getValue('foo'), 'bar', 'did set valid value')
+
+  const t2 = new TraceState(Buffer.from([]), 'es')
+  t2.setValue('foo', 'bar')
+  t2.setValue('zip', 'zap')
+  // now, set a key and value that are <257 characters, but that
+  // would result in the entire `es` value being greater than 256
+  const oneTwenty = new Array(120 + 1).join('x')
+  t.ok(!t2.setValue(oneTwenty, oneTwenty), 'call to setValue returned false when key too long')
+  t.equals(t2.getValue('foo'), 'bar', 'maintained value of foo key')
+  t.equals(t2.getValue('zip'), 'zap', 'maintained value of zip key')
+  t.ok(!t2.getValue(oneTwenty), 'did not set key/value that would put us over total character limit')
+
+  // same as t2 test, but oneTwenty key is previously set
+  const t3 = new TraceState(Buffer.from([]), 'es')
+  t3.setValue('foo', 'bar')
+  t3.setValue('zip', 'zap')
+  t3.setValue(oneTwenty, 'bees')
+  // now, set a key and value that are <257 characters, but that
+  // would result in the entire `es` value being greater than 256
+
+  t3.setValue(oneTwenty, oneTwenty)
+  t.equals(t3.getValue('foo'), 'bar', 'maintained value of foo key')
+  t.equals(t3.getValue('zip'), 'zap', 'maintained value of zip key')
+  t.equals(t3.getValue(oneTwenty), 'bees', 'maintained value of zip key')
+
+  t.end()
+})
+
+test('TraceState delimiter edge cases', function (t) {
+  // makes sure weird stuff doesn't blow anything up
+  const stringFormats = [
+    'foo=34f067aa0ba902b7,bar=0.25,es=a:b;cee:de,34@ree=xxxy',
+    'foo=34f067aa0ba902b7,bar=0.25,es=a:b;cee:de,34@ree=xxxy,',
+    'foo=34f067aa0ba902b7,bar=0.25,es=a:b;cee:de,;34@ree=xxxy,',
+    'foo=34f067aa0ba902b7,bar=0.25,es=a:b;cee:de;,34@ree=xxxy,',
+    'foo=34f067aa0ba902b7,,bar=0.25,es=a:b;;cee:de,34@ree=xxxy;',
+    'foo=34f067aa,0ba902b7,,bar=0.25,es=a:b;;cee:de,34@ree=xxxy;',
+    'foo=34f067aa,0ba9,02b7,,bar=0.25,es=a:b;;;cee:de,,,34@ree=xxxy;',
+    'foo=34f067aa0ba902b7,bar=0.25,es=a::b;cee:de,34@ree=xxxy',
+    'foo=34f067aa0ba902b7,bar=0.25,es=a:b;cee:de,es=xxxy'
+  ]
+  for (const [, format] of stringFormats.entries()) {
+    this.ok(TraceState.fromStringFormatString(format))
```

Comment:
nit: `t.ok`

Reply:
Looks like there's more than on occurrence of this :facepalm: -- surprised that `this.` actually works -- I guess `tape` binds `this` to -- a? the current? -- test.  

Good catch -- better to be explicit here.   This is fixed now, all tests use `t` object.